modules/apps/app-builder/app-builder-web/src/main/resources/META-INF/resources/js/hooks/useDataDefinition.es.js

```diff
@@ -16,13 +16,17 @@ import {getItem} from 'data-engine-js-components-web/js/utils/client.es';
 import {getLocalizedValue} from 'data-engine-js-components-web/js/utils/lang.es';
 import {useEffect, useState} from 'react';
 
-export default function useDataDefinition(dataDefinitionId) {
-	const [dataDefinition, setDataDefinition] = useState({});
+export default function useDataDefinition({
+	dataDefinitionId,
+	defaultState = {},
+	queryFields = '',
```

Comment:
Hey @kevenleone, if we make a request without **queryFields**, it will be `/o/data-engine/v2.0/data-definitions/123?fields=''`. Is it working? Even if it works, could we check if there are no **queryFields** and change the url? like:
`/o/data-engine/v2.0/data-definitions/123`

Reply:
I tested without passing any queryFields and worked fine.

But i changed the implementation, if you could see in my last commit, i think is better now.

If everything is ok, is good to go!

Reply:
@interaminense